internal/app/db/storage/postgres/order.go

```diff
@@ -0,0 +1,89 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log/slog"
+
+	"github.com/AsakoKabe/gophermart/internal/app/db/models"
+)
+
+type OrderStorage struct {
+	db *sql.DB
+}
+
+func NewOrderStorage(db *sql.DB) *OrderStorage {
+	return &OrderStorage{db: db}
+}
+
+const insertOrder = "insert into orders (num, user_id) values ($1, $2)"
+const selectOrderByNum = "select id, num, user_id, trim('\"' from to_json(uploaded_at)::text) from orders where num = $1"
+const selectOrdersByUserID = "select id, num, user_id, trim('\"' from to_json(uploaded_at)::text) from orders where user_id = $1 order by uploaded_at"
+
+func (s *OrderStorage) Add(ctx context.Context, order *models.Order) error {
+	_, err := s.db.ExecContext(ctx, insertOrder, order.Num, order.UserID)
+	if err != nil {
+		return fmt.Errorf("unable to insert new user: %w", err)
+	}
+
+	return nil
+}
+
+func (s *OrderStorage) GetOrderByNum(ctx context.Context, num string) (*models.Order, error) {
+	rows, err := s.db.QueryContext(
+		ctx,
+		selectOrderByNum,
+		num,
+	)
```

Comment:
```suggestion
	row := s.db.QueryRowContext(
		ctx,
		selectOrderByNum,
		num,
	)
```
А дальше можно сразу
```
order, err := s.parseOrder(row)
```